### PR TITLE
disallow uploading from `rattler-build` feedstocks

### PR DESCRIPTION
**Pull request by wolfv**
Also includes a small refactor to make the `determine_build_tool` function reusable.

### CODE REVIEW

File: recipe/conda_forge_ci_setup/build_utils.py [hunk 1]
```diff
@@ -228,6 +228,18 @@ def upload_package(feedstock_root, recipe_root, config_file, validate, private,
...
 
+    build_tool = determine_build_tool(feedstock_root)
+    if build_tool != CONDA_BUILD and upload_to_conda_forge:
+        # make sure that we are not uploading to the main conda-forge channel
+        # when building packages with `rattler-build`
+        if any(["label/" not in channels]):
+            print(
+                "Uploading to conda-forge's main channel is not yet allowed when building with rattler-build\n"
+                "You can set a label channel in the channel_targets section of the config file\n"
+                "to upload to a label channel"
```
Comment (beckermr) [inline recipe/conda_forge_ci_setup/build_utils.py:239]:
```suggestion
                "Uploading to conda-forge's main channel is not yet allowed when building with rattler-build.\n"
                "You can set a label channel in the channel_targets section of the config file\n"
                "to upload to a label channel."
```

**Review by beckermr — APPROVED** [review]

**Review by beckermr — CHANGES_REQUESTED** [review]
Woops! You need to bump the package version in the recipe.

File: recipe/conda_forge_ci_setup/build_utils.py [hunk 2]
```diff
@@ -228,6 +228,18 @@ def upload_package(feedstock_root, recipe_root, config_file, validate, private,
                     "is not allowed" % ("conda-forge", source_channel))
                 return
 
+    build_tool = determine_build_tool(feedstock_root)
+    if build_tool != CONDA_BUILD and upload_to_conda_forge:
+        # make sure that we are not uploading to the main conda-forge channel
+        # when building packages with `rattler-build`
+        if any(["label/" not in channels]):
```
Comment (isuruf) [inline recipe/conda_forge_ci_setup/build_utils.py:235]:
```suggestion
        if any(["label/" not in channel for channel in channels]):
```

**Comment by wolfv** [comment]
I bumped the version and use a generator expression directly plus also checking that the owner is conda-forge.

File: recipe/conda_forge_ci_setup/build_utils.py [hunk 3]
```diff
@@ -228,6 +228,18 @@ def upload_package(feedstock_root, recipe_root, config_file, validate, private,
                     "is not allowed" % ("conda-forge", source_channel))
                 return
 
+    build_tool = determine_build_tool(feedstock_root)
+    if build_tool != CONDA_BUILD and upload_to_conda_forge:
+        # make sure that we are not uploading to the main conda-forge channel
+        # when building packages with `rattler-build`
+        if any(owner == "conda-forge" and label == "main" for owner, label in channels):
```
Comment (isuruf) [inline recipe/conda_forge_ci_setup/build_utils.py:235]: :smile: 
```suggestion
        if ["conda-forge", "main"] in channels:
```

File: recipe/conda_forge_ci_setup/build_utils.py [hunk 4]
```diff
@@ -228,6 +228,18 @@ def upload_package(feedstock_root, recipe_root, config_file, validate, private,
...
+    build_tool = determine_build_tool(feedstock_root)
+    if build_tool != CONDA_BUILD and upload_to_conda_forge:
+        # make sure that we are not uploading to the main conda-forge channel
+        # when building packages with `rattler-build`
+        if ["conda-forge", "main"] in channels:
+            print(
+                "Uploading to conda-forge's main channel is not yet allowed when building with rattler-build.\n"
+                "You can set a label channel in the channel_targets section of the config file\n"
+                "to upload to a label channel."
+            )
```
Comment (jakirkham) [inline recipe/conda_forge_ci_setup/build_utils.py:240]: Wonder if we should open an issue to track discussion on this test rollout so we can evaluate when we are ready to enable uploads

**Comment by jakirkham** [comment]
Looks like all the builds failed on `main`. For [example]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=964384&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&t=b2a8456a-fb11-5506-ca32-5ccd32538dc0&l=1323 ):

```
Failed to upload due to name 'RATTLER_BUILD' is not defined. Trying again in 10 seconds
```

Is this expected?

**Pull request merged**